src/components/CustomFileInput/index.tsx

```diff
@@ -29,15 +31,20 @@ export default class CustomFileInput extends Component {
     this.props.onClick()
   }
 
-  handleChange = (event) => {
-    if (this.input) {
+  handleChange = (event: Event) => {
+    if (this.input && this.input.files) {
       this.props.onChange(this.input.files[0])
     }
-    event.currentTarget.value = '' // Allow re-uplading the same file
+
+    if (event.currentTarget) {
+      const inputTarget = event.currentTarget as HTMLInputElement
+      inputTarget.value = '' // Allow re-uploading the same file
+    }
   }
 
   render = () => {
-    const { children, className, onClick, onChange, ...other } = this.props // eslint-disable-line no-unused-vars
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { children, className, onClick, onChange, ...other } = this.props
```

Comment:
We want to exclude them from `other`. Otherwise `<input>` complains that `onChange` is already defined, and `onClick` will trigger and it's not what is expected.

This is why I'm scared to use {...props} 😅